src/main/scala/com/amazon/deequ/analyzers/Analyzer.scala

```diff
@@ -289,6 +289,34 @@ object Preconditions {
 
   private[this] val nestedDataTypes = Set(StructType, MapType, ArrayType)
 
+  private[this] val caseSensitive = {
+    val sparkSession = SparkSession.builder().getOrCreate()
+    try {
```

Comment:
Can we check this without relying on exceptions?

Reply:
Yes, it has been done.